src/method.jl

```diff
@@ -0,0 +1,114 @@
+"""
+    signature(meth::Method) -> Dict{Symbol,Any}
+
+Finds the expression for a methods signature as broken up into its various components
+including:
+
+- `:head`: Expression head of the function definition: always `:function`
+- `:name`: Name of the function (not present for anonymous functions)
+- `:params`: Parametric types defined on constructors
+- `:args`: Positional arguments of the function
+- `:rtype`: Return type of the function
+- `:whereparams`: Where parameters
+
+All components listed may not be present in the returned dictionary with the exception of
+`:head` which will always be present.
+
+Note: keyword arguments are always ignored, right now.
+
+These are the same components returned by [`splitdef`](@ref) and consumed by
+[`combinedef`](@red), except for the `:body` component which will never be present.
+"""
+function signature(meth::Method)
+    def = Dict{Symbol,Any}()
+    def[:head] = :function
+    def[:name] = meth.name
+
+
+    def[:args] = get_args(meth)
+    def[:whereparams] = get_whereparams(meth)
+
+    return Dict(k=>v for (k, v) in def if v !== nothing)  # filter out nonfields.
+end
+
+function get_slot_sym(meth::Method)
+    # In 1.3 can do:
+    # Symbol.(split(meth.slot_syms, '\0'; keepempty=false))
+    ci = Base.uncompressed_ast(meth)
+    return ci.slotnames
+end
+function get_arg_names(meth::Method)
+    slot_syms = get_slot_sym(meth)
+    @assert slot_syms[1] == Symbol("#self#")
+    arg_names = slot_syms[2:meth.nargs]  #nargs includes 1 for #self#
+    return arg_names
+end
+
+"""
+    parameters(type)
+
+extracts the type-parameters of the `type`.
+E.g. `parameters(Foo{A, B, C}) == [A, B, C]`
```

Comment:
```suggestion
```jldoctest
julia> parameters(Foo{A, B, C}) == [A, B, C]
true
```

Reply:
This, and all the other private functions in this file need `_` prefixes